### ISSUE TITLE
Create a tutorial that shows how to query multiple reaches in parallel.

### ISSUE THREAD

**Issue by nikki-t** [podaac/hydrocron#128]
Create a tutorial for the PO.DAAC cookbook that demonstrates how to query multiple reaches in parallel.

**Comment by cassienickles** [comment]
Thanks Nikki! I can take care of all the logistics of formatting the tutorial & putting it in the Cookbook & documentation, but I will need your help on the process/code to actually do this.

**Comment by cassienickles** [comment]
Started template for tutorial on a 'Hydrocron_example' branch of the tutorials repo here: https://github.com/podaac/tutorials/blob/Hydrocron_example/notebooks/datasets/Hydrocron_SWOT_timeseries_examples.ipynb

**Comment by cassienickles** [comment]
For using FTS to obtain reach IDs, I'm stuck on getting the results from the next page of responses, there's a cap of 100 results per page and thus far 30 minutes of Googling has not yielded working results... would appreciate insight if you already know how to do this when you work on it next week, Nikki!

**Comment by nikki-t** [comment]
Can you share the command you are using to query FTS by reach ID? Then I can play around with it!

**Comment by cassienickles** [comment]
![image](https://github.com/podaac/hydrocron/assets/44206543/f8540bf2-70a4-4c89-baf2-a26e62630ce3)

Savannah_reaches = requests.get("https://fts.podaac.earthdata.nasa.gov/rivers/reach/732520")
pprint.pprint(Savannah_reaches.json(), compact=True, width=60, depth=2)

# This is only a partial result from page 1, how do we also append from the rest of the pages in the response?
IDs = list(Savannah_reaches.json()['results'].keys())

**Comment by nikki-t** [comment]
@cassienickles - I have [updated the notebook ](https://github.com/podaac/tutorials/blob/Hydrocron_example/notebooks/datasets/Hydrocron_SWOT_timeseries_examples.ipynb)with a rough sketch of using Dask to query Hydrocron by reach id in parallel (which were retrieved from FTS by river name).

You can ignore the "Shuffle" warnings for now. I also couldn't figure out how to filter out missing data on the Dask DataFrame but was able to do so using a regular Pandas DataFrame. So I would like to explore that more as the plotting library I used works well with Dask.

We could also remove Dask as I am not sure how much it is buying us unless we wanted to try to load in hundreds of reach identifiers which I think I will try doing!!

Here is the final plot that gets created. 
![Screenshot 2024-04-22 at 16 43 22](https://github.com/podaac/hydrocron/assets/17799906/9cc764a3-e728-44e7-90ca-0d37bdee670b)

Totally open to feedback on any modifications to the tutorial and how to visualize the multiple reach time series!

**Comment by cassienickles** [comment]
I like your thought process, Nikki! I added some markdown explanations and made it flow a bit more like a tutorial and committed back to the branch! 

One thing I notice right away, the dates are being plotted as strings rather than datetime format so the time series looks pretty wonky. I tried to go in and fix it but I don't know the correct way to do this for dask dataframe. I tried what I usually would for a pandas dataframe:

ddf.time_str = pd.to_datetime(ddf.time_str)

and get an error that the length of values don't match... Thoughts? How do we fix this with a dask dataframe? 

I think the benefit of dask would be to load in hundreds of reachIDs, would love to hear the results if you mess around with this.
